Remotion/Web/ClientScript/Scripts/DropDownMenu.ts

```diff
@@ -0,0 +1,928 @@
+
+      $(anchor).attr(itemInfo.DiagnosticMetadataForCommand);
+    }
+
+    var span = document.createElement('span');
+    span.innerHTML = itemInfo.Text;
+    anchor.appendChild(span);
+    $(anchor)
+      .bind ('focus', DropDownMenu.FocusHandler)
+      .bind ('blur', DropDownMenu.BlurHandler);
+
+    return item;
+  }
+
+  private static CreateSeparatorItem(): HTMLLIElement
+  {
+    var item = document.createElement('li');
+
+    var textPane = document.createElement('span');
+    textPane.className = DropDownMenu._itemSeparatorClassName;
+    item.setAttribute('role', 'none');
+    textPane.innerHTML = '&nbsp;';
+
+    item.appendChild(textPane);
+
+    return item;
+  }
+
+  public static OnHeadControlClick(): void
+  {
+    DropDownMenu._itemClicked = true;
+  }
+
+  public static OnKeyDown(event: JQueryKeyEventObject, dropDownMenu: JQuery, getSelectionCount: Nullable<() => number>, hasDedicatedDropDownMenuElement: boolean): void
+  {
+    ArgumentUtility.CheckNotNullAndTypeIsJQuery('dropDownMenu', dropDownMenu);
+
+    if (DropDownMenu._currentMenu === null && !hasDedicatedDropDownMenuElement)
+      return;
+
+    switch (event.keyCode)
+    {
+      case 9: // tab
+        DropDownMenu.ClosePopUp(DropDownMenu._updateFocus);
+        return;
+      case 13: //enter
+      case 32: //space
+        event.preventDefault();
+        event.stopPropagation();
+        if (dropDownMenu !== DropDownMenu._currentMenu)
+        {
+          DropDownMenu.OnClick (dropDownMenu, dropDownMenu[0].id, getSelectionCount, null);
+          event.keyCode = 40; // always act as if the down-arrow was used when opening the drop down menu.
+          DropDownMenu.Options_OnKeyDown (event, dropDownMenu);
+        }
+        else
+        {
+          DropDownMenu.ClosePopUp(DropDownMenu._updateFocus);
+        }
+        return;
+      case 27: //escape
+        event.preventDefault();
+        event.stopPropagation();
+        DropDownMenu.ClosePopUp(DropDownMenu._updateFocus);
+        return;
+      case 38: // up arrow
+      case 40: // down arrow
+        event.preventDefault();
+        event.stopPropagation();
+        if (dropDownMenu !== DropDownMenu._currentMenu)
+        {
+          DropDownMenu.OnClick (dropDownMenu, dropDownMenu[0].id, getSelectionCount, null);
+          event.keyCode = 40; // always act as if the down-arrow was used when opening the drop down menu.
+        }
+        DropDownMenu.Options_OnKeyDown (event, dropDownMenu);
+        return;
+      default:
+        DropDownMenu.Options_OnKeyDown(event, dropDownMenu);
+        return;
+    }
+  }
+
+  private static Options_OnKeyDown(event: JQueryKeyEventObject, dropDownMenu: JQuery): void
+  {
+    ArgumentUtility.CheckNotNullAndTypeIsJQuery('dropDownMenu', dropDownMenu);
+
+    if (DropDownMenu._currentPopup == null)
+      return;
+
+    // TODO: bug?
+    var itemInfos = (DropDownMenu._menuInfos[dropDownMenu[0].id] as DropDownMenu_MenuInfo).ItemInfos;
```

Comment:
```suggestion
    var itemInfos = DropDownMenu._menuInfos[dropDownMenu[0].id]!.ItemInfos!;
```

We should not split up the same todo into multiple smaller parts. We can use a `!` at the end to do it once. This is then removed once the bug is fixed. Any usages do not need any further annotation